flink-runtime/src/main/java/org/apache/flink/runtime/leaderelection/DefaultLeaderElectionService.java

```diff
@@ -268,9 +272,13 @@ public void close() throws Exception {
                 running = false;
             } else {
                 LOG.debug("The HA backend connection isn't established. No actions taken.");
+                return;
             }
         }
 
+        Preconditions.checkState(
+                leaderElectionDriver != null, "The HA backend wasn't initialized.");
+
         leaderElectionDriver.close();
```

Comment:
isnt it possible for a service to be closed without a contender ever being registered (and thus the driver being null)?

Reply:
In that case, `running` is `false` and we wouldn't reach this part of the code. I created FLINK-32504 to improve the lifecycle management of this class. Right now, it's a bit confusing because we have the `running` flag that serves as a guard. But we could get rid of it which would make it more transparent, I guess, when we would reach what part of the code.